chainer/functions/array/where.py

```diff
@@ -26,11 +26,13 @@ def forward(self, inputs):
         condition, x, y = inputs
         return xp.where(condition, x, y),
 
-    def backward(self, inputs, grads):
-        xp = cuda.get_array_module(inputs[0])
-        condition = inputs[0]
-        gx = xp.where(condition, grads[0], 0)
-        gy = xp.where(condition, 0, grads[0])
+    def backward(self, indexes, grad_outputs):
+        condition = self.inputs[0]
+        xp = cuda.get_array_module(condition.data)
+        g, = grad_outputs
+        zeros = xp.zeros(g.shape, dtype=g.dtype)
+        gx, = Where().apply((condition.data, g, zeros))
```

Comment:
Could you skip the computation of `gx`/`gy` if not needed? Whether or not they are needed can be determined by `indexes`.

Reply:
Fixed.